apps/web/components/eventtype/MinimumBookingNotice.tsx

```diff
@@ -0,0 +1,96 @@
+import React, { useEffect, useState } from "react";
+import type { UseFormRegisterReturn } from "react-hook-form";
+import { useFormContext } from "react-hook-form";
+
+import type { FormValues } from "@calcom/features/eventtypes/lib/types";
+import type { DurationType } from "@calcom/lib/convertToNewDurationType";
+import convertToNewDurationType from "@calcom/lib/convertToNewDurationType";
+import findDurationType from "@calcom/lib/findDurationType";
+import { useLocale } from "@calcom/lib/hooks/useLocale";
+import { InputField, Select } from "@calcom/ui";
+
+type MinimumBookingFields = "minimumBookingNotice" | "seatsMinimumBookingNotice";
+
+// Define a generic type for the props, allowing dynamic passing of the `name` field
+type MinimumBookingNoticeInputProps<T extends MinimumBookingFields> = {
+  name: T;
+} & Omit<UseFormRegisterReturn<T>, "ref">;
+
+export const MinimumBookingNoticeInput = React.forwardRef<
+  HTMLInputElement,
+  MinimumBookingNoticeInputProps<MinimumBookingFields>
+>(function MinimumBookingNoticeInput({ name, ...passThroughProps }, ref) {
+  const { t } = useLocale();
+  const { setValue, getValues } = useFormContext<FormValues>();
+  const durationTypeOptions: {
+    value: DurationType;
+    label: string;
+  }[] = [
+    { label: t("minutes"), value: "minutes" },
+    { label: t("hours"), value: "hours" },
+    { label: t("days"), value: "days" },
+  ];
+
+  const initialValue = getValues(name) ?? 0;
```

Comment:
getValues(name) can be null, if name = "seatsMinimumBookingNotice". 
Todo: Need to handle that behaviour explicitly